Applications/Spire/Include/Spire/Ui/DropDownBoxAdaptor.hpp

```diff
@@ -0,0 +1,140 @@
+#ifndef SPIRE_DROP_DOWN_BOX_ADAPTOR_HPP
+#define SPIRE_DROP_DOWN_BOX_ADAPTOR_HPP
+#include <QHBoxLayout>
+#include <QWidget>
+#include "Spire/Ui/DropDownBox.hpp"
+#include "Spire/Ui/ListView.hpp"
+#include "Spire/Ui/Ui.hpp"
+
+namespace Spire {
+
+  /**
+   * Implements a wrapper over a DropDownBox for use with a type.
+   * @param <T> The type of value to display in the DropDownBox.
+   */
+  template<typename T>
+  class DropDownBoxAdaptor : public QWidget {
+    public:
+
+      /** The type of the values used to display in the DropDownBox. */
+      using Type = T;
+
+      /**
+       * Signals that the value was submitted.
+       * @param submission The submitted value.
+       */
+      using SubmitSignal = Signal<void (const Type& submission)>;
+
+      /**
+       * Constructs a DropDownBoxAdaptor using a LocalValueModel.
+       * @param list_model The model of values to display.
+       * @param parent The parent widget.
+       */
+      DropDownBoxAdaptor(std::shared_ptr<ListModel> list_model,
+        QWidget* parent = nullptr);
+
+      /**
+       * Constructs a DropDownBoxAdaptor.
+       * @param list_model The model of values to display.
+       * @param current_model The current value's model.
+       * @param parent The parent widget.
+       */
+      DropDownBoxAdaptor(std::shared_ptr<ListModel> list_model,
+        std::shared_ptr<ValueModel<boost::optional<int>>> current_model,
+        QWidget* parent = nullptr);
+
+      /** Returns the list model. */
+      const std::shared_ptr<ListModel>& get_list_model() const;
+
+      /** Returns the current model. */
+      const std::shared_ptr<ValueModel<boost::optional<int>>>&
+        get_current_model() const;
+
+      /** Returns <code>true</code> iff this DropDownBoxAdaptor is read-only. */
+      bool is_read_only() const;
+
+      /**
+       * Sets the read-only state.
+       * @param is_read_only True iff the DropDownBoxAdaptor should be
+       *        read-only.
+       */
+      void set_read_only(bool is_read_only);
+
+      /** Connects a slot to the submit signal. */
+      boost::signals2::connection connect_submit_signal(
+        const typename SubmitSignal::slot_type& slot) const;
+
+      private:
+        mutable SubmitSignal m_submit_signal;
+        std::shared_ptr<ListModel> m_list_model;
```

Comment:
This suggests that the ```ListModel```'s ownership is shared by the ```DropDownBoxAdaptor``` and the ```ListView```, why is this shared with both instead of having the ```ListView``` own it?